src/main/helpers.js

```diff
@@ -290,3 +290,49 @@ export function readableStream(data) {
   s.push(null)
   return s
 }
+// Function prepends metadata with the appropriate prefix if it is not already on
+export function prependXAMZMeta(metaData) {
+  var newMetadata = Object.assign({}, metaData)
+  for (var key in metaData) {
+    if(!isAmzHeader(key) && !isSupportedHeader(key) && !isStorageclassHeader(key)) {
+      newMetadata["X-Amz-Meta-" + key ] = newMetadata[key]
+      delete newMetadata[key]
+    }
+  }
+  return newMetadata
+}
+
+// Checks if it is a valid header according to the AmazonS3 API
+export function isAmzHeader(key) {
+  var temp = key.toLowerCase()
+  return temp.startsWith("x-amz-meta") || temp === "x-amz-acl" || temp.startsWith("x-amz-server-side-encryption")
```

Comment:
Can this be changed to `return temp.startsWith("x-amz-meta-") `  ?

Reply:
fixed